src/main/java/com/bandwidth/sqs/client/SqsAsyncIoClient.java

```diff
@@ -8,22 +8,27 @@
 import com.amazonaws.services.sqs.model.DeleteMessageBatchResult;
 import com.amazonaws.services.sqs.model.DeleteMessageRequest;
 import com.amazonaws.services.sqs.model.DeleteMessageResult;
+import com.amazonaws.services.sqs.model.Message;
 import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
 import com.amazonaws.services.sqs.model.ReceiveMessageResult;
 import com.amazonaws.services.sqs.model.SendMessageBatchRequest;
 import com.amazonaws.services.sqs.model.SendMessageBatchResult;
 import com.amazonaws.services.sqs.model.SendMessageRequest;
 import com.amazonaws.services.sqs.model.SendMessageResult;
+import com.bandwidth.sqs.publisher.MessagePublisher;
+
+import java.time.Duration;
+import java.util.Optional;
 
 import io.reactivex.Single;
 
-public interface SqsAsyncIoClient {
+public interface SqsAsyncIoClient extends MessagePublisher<Message>{
 
     Single<ReceiveMessageResult> receiveMessage(ReceiveMessageRequest request);
 
     Single<DeleteMessageResult> deleteMessage(DeleteMessageRequest request);
 
-    Single<SendMessageResult> sendMessage(SendMessageRequest request);
+    Single<SendMessageResult> publishMessage(Message message, String queueUrl, Optional<Duration> delay);
```

Comment:
Mainly because I wanted to implement `MessagePublisher` which is generic, so the "Message" part can change while the queueUrl/delay should stay the same. This becomes a mess with `SendMessageRequest`. Also, `SendMessageRequest` has several options that will never be supported by this library. This library only supports "at-least once" queues, where SQS recently added "exactly-once queues" and just combined the apis in a very confusing and messy way.